ansible_base/authentication/serializers/authenticator_map.py

```diff
@@ -17,17 +25,69 @@ def validate(self, data) -> dict:
         map_type = data.get('map_type', None)
         team = data.get('team', None)
         org = data.get('organization', None)
-        if map_type == 'team' and (not team or team == ''):
-            errors["team"] = "You must specify a team with the selected map type"
-        if map_type == 'team' and (not org or org == ''):
-            errors["organization"] = "You must specify an organization with the selected map type"
-        if map_type == 'organization' and (not org or org == ''):
-            errors["organization"] = "You must specify an organization with the selected map type"
+        role = data.get('role', None)
+
+        if map_type == 'team' and not team:
+            errors["team"] = _("You must specify a team with the selected map type")
+        if map_type in ['team', 'organization'] and not org:
+            errors["organization"] = _("You must specify an organization with the selected map type")
+        if map_type in ['team', 'organization', 'role'] and not role:
+            errors["role"] = _("You must specify a role with the selected map type")
+        if map_type in ['allow', 'is_superuser'] and role:
+            errors["role"] = _("You cannot specify role with the selected map type")
+
+        if role:
+            errors.update(self.validate_role_data(map_type, role, org, team))
 
         if errors:
             raise ValidationError(errors)
         return data
 
+    def validate_role_data(self, map_type, role, org, team):
+        errors = {}
+
+        # Validation is possible only if RBAC is installed
+        if not apps.is_installed('ansible_base.rbac'):
+            logger.warning(_("You specified a role without RBAC installed "))
+            return errors
```

Comment:
Somewhere in the Django documentation I found a note that INSTALLED_APPS shouldn't be used directly (but not mentioned why)

Reply:
so, I guess you were not talking about the format of `if` condition, right? 

To the meaning, we agreed that authentication app doesn't need RBAC to be installed (at least now). 
So the condition in `dynamic_settings.py` is not applicable, because inside of this condition it's using the RBAC models, which are optional